offer/src/main/java/org/shareio/backend/infrastructure/controller/OfferRESTController.java

```diff
@@ -51,6 +52,43 @@ public ResponseEntity<Object> getConditions() {
         return new CorrectResponse(new ConditionsResponseDto(conditionsWithDisplayNames), Const.successErrorCode, HttpStatus.OK);
     }
 
+    @RequestMapping(value = "/getClosestOfferForUser/{userId}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Object> getClosestOfferForUser(@PathVariable(value = "userId") UUID userId) {
+        UserProfileResponseDto userProfileResponseDto;
+        try {
+            userProfileResponseDto = getUserProfileUseCaseInterface.getUserProfileResponseDto(userId);
+        } catch (MultipleValidationException e) {
+            return new ErrorResponse(e.getErrorMap(), e.getMessage(), HttpStatus.FAILED_DEPENDENCY);
+        } catch (NoSuchElementException noSuchElementException) {
+            return new ErrorResponse(Const.noSuchElementErrorCode, HttpStatus.NOT_FOUND);
+        }
+
+        //TODO Move location validation to UserValidator
+        Optional<LocationGetDto> locationGetDto = getLocationDaoInterface.getLocationDto(userProfileResponseDto.address().getId());
+        if (locationGetDto.map(Location::fromDto).isPresent()) {
+            UUID closestOfferId = getClosestOfferUseCaseInterface.getOfferResponseDto(locationGetDto.map(Location::fromDto).get());
+            return new CorrectResponse(closestOfferId, Const.successErrorCode, HttpStatus.OK);
+        } else {
+            return new ErrorResponse(Const.toDoErrorCode, HttpStatus.FAILED_DEPENDENCY);
+        }
+
+    }
+
+    @RequestMapping(value = "/getOffersByName", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Object> debugGetOffersByName(@RequestParam String name) {
```

Comment:
If this is a debug endpoint, it should be in DebugController. If this is a real production endpoint, it shouldn't be called debug